### PR TITLE
Give item feature

### PR DESCRIPTION
**Pull request by George-Hachem**
The owner of an item is now able to give it to the user. The owner of a post is able to see a Give button in the modal overlay. Once pressed, it will take the owner to /give page carrying the postId of the post with it. Once in the /give page, the owner will then be able to search for a user to give the item to. Once the owner presses submit, the available and userAcquired attributes in the post model will change to false and userId(the user that acquires the item) respectively.

### CODE REVIEW

**Review by AmineSlaoui — COMMENTED** [review]
Comments?

**Comment by George-Hachem** [comment]
Added comments

File: frontend/src/pages/give/give.jsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            navigate('/feed');
+
+        }catch(error){
+            console.error('Error giving the item:', error.response ? error.response.data : error);
+            alert('Failed to give the item. Please try again.');
+        }
+    }
+    return (
+        <>
+        <div className="serach-content-conatiner">
```
Comment (AmineSlaoui) [inline frontend/src/pages/give/give.jsx:82]: serach ?

Comment (George-Hachem) [reply frontend/src/pages/give/give.jsx]: Fixed it

**Review by AmineSlaoui — APPROVED** [review]
Looks good, is good

**Review by IanMcC123 — APPROVED** [review]

**Pull request merged**